apisix/core/etcd.lua

```diff
@@ -15,11 +15,38 @@
 -- limitations under the License.
 --
 local fetch_local_conf = require("apisix.core.config_local").local_conf
-local etcd = require("resty.etcd")
-local clone_tab = require("table.clone")
+local etcd             = require("resty.etcd")
+local clone_tab        = require("table.clone")
+local io               = io
+local type             = type
+local ipairs           = ipairs
+local string           = string
+local tonumber         = tonumber
 
 local _M = {version = 0.1}
 
+local prefix_v3 = {
+    ["3.5"] = "/v3",
+    ["3.4"] = "/v3",
+    ["3.3"] = "/v3beta",
+    ["3.2"] = "/v3alpha",
+}
+
+
+-- TODO: Default lua-resty-etcd version auto-detection is broken, so directly get version from cmd
+--          we don't need to call this so many times, need to save it in some place
+local function etcd_version_from_cmd()
+    local cmd = "export ETCDCTL_API=3 && etcdctl version"
+    local t, err = io.popen(cmd)
+    if not t then
+        return nil, "failed to execute command: " .. cmd .. ", error info:" .. err
+    end
+    local data = t:read("*all")
+    t:close()
+    return prefix_v3[data:sub(-4,-2)]
+end
+_M.etcd_version_from_cmd = etcd_version_from_cmd
```

Comment:
if `etcd_version_from_cmd` failed, `_M.etcd_version_from_cmd` will be `nil`, is test case cover this?

Reply:
Thank you!
fixed